accepted/00000-ansible-integration.md

```diff
@@ -0,0 +1,304 @@
+  * This would require, of course UI and DB investment. The idea would be to maintain the Ansible catalog inside the Uyuni server and push the playbook directly to the controller or the Ansible systems.
+  * Some questions: Which inventory / hosts groups and Ansible controller to use here for running the playbooks? Uyuni server as the Ansible controller?
+
+- Ansible Playbooks with Forms
+  * Similarly to what we currently have for Salt with "Formulas with Forms". Prefilled playbooks + some metadata to render the forms to filling the required information. This rendered playbook can be exposed via Salt fileserver as described above so the corresponding controller is able to fetch it.
+
+
+#### New UI tabs & pages
+
+- TBD
+
+# Defining a MVP
+[mvp]: #mvp
+
+### Step 1: Uyuni as your Ansible controller
+- Ansible is installed on the Uyuni server.
+- Ansible inventory, hosts are manually defined in the Uyuni server.
+- The user takes care of provide an Ansible SSH key which is accepted in the hosts from the Ansible inventory.
+- Inventory can be synced with Uyuni (add "Foreign/ANSIBLE" and "Salt/ANSIBLE", ...)
+- The user can operate Ansible via CLI.
+- Transition from "Foreign/ANSIBLE" to "Salt/ANSIBLE".
+
+### Step 2: Enhancing the UI
+- Playbook catalog
+- Remote commands for "Foreign/ANSIBLE" systems.
+- Improve "Visualization" features.
+
+### Step 3: Support multiple Ansible controller
+- Add "ansible-gatherer" to deal with multiple Ansible controllers
+- New UI pages and DB changes to deal with different controllers
+- Enhance "Playbook catalog" for multiple Ansible controllers.
+
+### Step 4:
+- Whatever comes next
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+Allowing Ansible clients in Uyuni sounds great, but at the same time, we need to think that Uyuni and its features are really based and tied to Salt. Allowing some basic integration, like collecting your Ansibles managed clients and expose them in Uyuni, operating your Ansible controller and some other things like easily migration to Salt minion are really cool and feasible featus, I think we should really think if we want to make Uyuni a tool that allows you to build, maintain and operate your Ansible infrastructure from scratch.
+
+Why should we **not** do this?
+
+  * obscure corner cases
+  * will it impact performance?
+  * what other parts of the product will be affected?
+  * will the solution be hard to maintain in the future?
+
+# Alternatives
+[alternatives]: #alternatives
+
+* Interfacing AWX.
+
+An alternative would be to use the API of AWX. While this would mean that all options of AWX would be available, it would also mean that we would need to make sure to always stay compatible with all versions used by users, breaking changes could be introduced by AWX and some features might even need to be implemented in AWX first before being able to use them in Uyuni.
```

Comment:
Do we have any data about the prevalence of AWX on the total number of Ansible users? If any significant portion uses bare Ansible, then we would also lose that part of the audience.

Reply:
I don't have a clear idea at the moment. Will try to figure out